diff --git a/node_modules/@vue/cli-service/lib/config/css.js b/node_modules/@vue/cli-service/lib/config/css.js
index 8ed83f5..1db58f3 100644
--- a/node_modules/@vue/cli-service/lib/config/css.js
+++ b/node_modules/@vue/cli-service/lib/config/css.js
@@ -49,7 +49,7 @@ module.exports = (api, rootOptions) => {
       requireModuleExtension = true
     }
 
-    const shouldExtract = extract !== false && !shadowMode
+    const shouldExtract = rootOptions.shouldExtractCss || extract !== false && !shadowMode
     const filename = getAssetPath(
       rootOptions,
       `css/[name]${rootOptions.filenameHashing ? '.[contenthash:8]' : ''}.css`
diff --git a/node_modules/@vue/cli-service/lib/options.js b/node_modules/@vue/cli-service/lib/options.js
index fad7a32..ff13b42 100644
--- a/node_modules/@vue/cli-service/lib/options.js
+++ b/node_modules/@vue/cli-service/lib/options.js
@@ -60,7 +60,9 @@ const schema = createSchema(joi => joi.object({
   pwa: joi.object(),
 
   // 3rd party plugin options
-  pluginOptions: joi.object()
+  pluginOptions: joi.object(),
+
+  shouldExtractCss: joi.boolean()
 }))
 
 exports.validate = (options, cb) => {
